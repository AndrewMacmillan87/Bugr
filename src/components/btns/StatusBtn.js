import React from "react";

import PropTypes from "prop-types";

import "./status-btn.css";

const StatusBtn = ({ id, value }) => {
    const classes = `issue-state-btn ${value}`;
    return (
        <button className={classes} id={id} type='button'>
            {value}
        </button>
    );
};

StatusBtn.propTypes = {
    id: PropTypes.string,
    value: PropTypes.string,
};

export default StatusBtn;
