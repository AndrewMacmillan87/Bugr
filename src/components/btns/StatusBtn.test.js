import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import StatusBtn from "./StatusBtn";

describe("StatusBtn component", () => {
    let props;
    beforeEach(() => {
        props = {
            id: "test",
            value: "Testing",
        };
    });
    test("Renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<StatusBtn {...props} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Renders correctly", () => {
        const tree = renderer.create(<StatusBtn {...props} />);
        expect(tree).toMatchSnapshot();
    });

    describe("StatusBtn HTML elements", () => {
        it("should render a button element", () => {
            const wrapper = mount(<StatusBtn {...props} />);
            expect(wrapper.find("button").length).toEqual(1);
        });
    });
});
