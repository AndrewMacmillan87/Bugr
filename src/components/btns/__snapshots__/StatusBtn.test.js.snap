// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBtn component Renders correctly 1`] = `
<button
  className="issue-state-btn Testing"
  id="test"
  onClick={[Function]}
  type="button"
>
  Testing
</button>
`;
